src/dmd/e2ir.d

```diff
@@ -260,7 +260,11 @@ private elem *callfunc(const ref Loc loc,
                 /* Copy to a temporary, and make the argument a pointer
                  * to that temporary.
                  */
-                elems[i] = addressElem(ea, arg.type, true);
+                VarDeclaration v = arg.op == TOK.variable ? (cast(VarExp)arg).var.isVarDeclaration() : null;
+                bool copy = !(v && v.isArgDtorVar); // copy unless the destructor is going to be run on it
+                                                    // then assume the frontend took care of the copying and pass it by ref
+
+                elems[i] = addressElem(ea, arg.type, copy);
```

Comment:
```
bool copy;
if (auto ve = arg.isVarExp())
{
    if (auto v = ve.isVarDeclaration())
        copy = v.isArgDtorVar;
}
```
I.e. please avoid casts.

Reply:
Oops, I have the logic inverted, but the idea is the same.

Reply:
done.